### ISSUE TITLE
Crash while quitting app with hidden title bar in fullscreen (Mac)

### ISSUE THREAD

**Issue by Honason** [atom/electron#3258]
When I try to quit an app that has BrowserWindow with 'title-bar-style': 'hidden' setting, and the app is in fullscreen at the time, this http://pastebin.com/Hxx8D9pi crash occurs. Works OK when quitting the app when it's not in fullscreen mode, or when this title-bar-style is not set.

Mac OSX 10.11.2 (same error was on 10.11.1), Electron v.0.34.1

**Comment by jlord** [comment]
Tried it on an app ([Git-it](https://github.com/jlord/git-it-electron)) with OS X 10.10.5 and Electron v.0.34.1 but couldn't reproduce.

Is there any app that you've seen this happen in that we can try with? Or are you able to make a simple demo app that reproduces it?

**Comment by Honason** [comment]
I've tried to clone this https://github.com/atom/electron/blob/master/docs/tutorial/quick-start.md quick start demo and added title-bar-style to BrowserWindow like this:
```javascript
mainWindow = new BrowserWindow({width: 800, height: 600, 'title-bar-style': 'hidden'});
```
which reproduced the error and threw the same error. 

Guessing from the error log, draggable regions could have something to do with this
```
atom::NativeWindowMac::UpdateDraggableRegionViews(std::__1::vector<atom::DraggableRegion, std::__1::allocator<atom::DraggableRegion> > const&) + 74
1   com.github.electron.framework       0x000000010c12ad30 atom::NativeWindowMac::UpdateDraggableRegionViews() + 16
```

**Comment by zcbenz** [comment]
I'm not able to reproduce too, but the crash report looks quite reasonable, probably it is 10.11 only or involves some race conditions.

**Comment by zcbenz** [comment]
Fixed with #3588.

**Comment by bengotow** [comment]
We've had a Nylas N1 user report this happening again on MacOS Sierra with Electron 1.1—we're going to continue investigating and we'll see if it's a regression.
